src/components/Contact/contactlist.tsx

```diff
@@ -1,13 +1,22 @@
 import { XCircleIcon } from '@heroicons/react/24/outline'
+import { useAtomValue } from 'jotai'
 
+import { searchQueryAtom, searchTagsAtom } from '@/atoms/contacts'
 import ContactItem from '@/components/Contact/contactitem'
 import { useContacts } from '@/hooks/contacts'
 
 const ContactList = () => {
   const { data: contacts } = useContacts()
+  const searchQuery = useAtomValue(searchQueryAtom)
+  const searchTags = useAtomValue(searchTagsAtom)
   if (contacts === undefined) return null
 
   const contactItems = contacts
+    .filter(
+      (contact) =>
```

Comment:
You can chain `filter`, I think for readability you should use a separate filter for the query and the tags.